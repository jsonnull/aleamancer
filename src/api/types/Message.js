// @flow

const Message = `
  type Message {
    id: ID!
    from: String!
    result: String
    text: String!
    timestamp: String!
  }

  extend type Query {
    message: Message
  }

  input MessageInput {
    from: String!
    result: String
    text: String!
  }

  extend type Mutation {
    sendMessage(message: MessageInput!): Message
  }
`

export default Message
