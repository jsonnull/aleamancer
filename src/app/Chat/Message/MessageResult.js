/* @flow */
import React from 'react'
import type { MessageResult } from 'types'
import styles from './style.css'

type RollsProps = {
  rolls: Array<number>
}
const Rolls = (props: RollsProps) => {
  return <span>
    <i className='fa fa-cube'></i>
    {' '}
    {props.rolls.map((roll, i, arr) => (
      <span key={i}>
        <span className={styles.roll}>
          { roll }
        </span>
        {(i + 1 !== arr.length) ? ' + ' : null}
      </span>
    ))}
  </span>
}

type ModProps = {
  mod: number
}
const Mod = (props: ModProps) => {
  if (props.mod == 0) {
    return null
  }

  return <span>
    {' + '}
    <span className={styles.mod}>{props.mod}</span>
  </span>
}

type Props = {
  result: ?MessageResult
}
const Result = (props: Props) => {
  if (!props.result) {
    return null
  }

  let { mod, rolls, total } = props.result

  return (
    <div className={styles.result}>
      <Rolls rolls={rolls} />
      <Mod mod={mod} />
      {' = '}
      <span className={styles.total}>{total}</span>
    </div>
  )
}

export default Result
