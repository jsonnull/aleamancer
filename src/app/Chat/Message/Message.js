/* @flow */
import React from 'react'
import type { Message } from 'types'
import MessageResult from './MessageResult'
import Timeago from 'timeago.js'
import styles from './style.css'

type Props = {
  message: Message
}

export default class MessageView extends React.Component<*, Props, *> {
  render () {
    const { from, timestamp, text, result } = this.props.message
    return <div className={styles.message}>
      <div className={styles.from}>
        { from }{' '}
        <span className={styles.date} title={ Date(timestamp) }>
          { new Timeago().format(timestamp) }
        </span>
      </div>
      <div className={styles.text}>
        { text }
      </div>
      <MessageResult result={result} />
    </div>
  }
}
