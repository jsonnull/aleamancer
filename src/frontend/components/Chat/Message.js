// @flow
import React from 'react'
import styled from 'styled-components'
import Timeago from 'timeago.js'
import type { Message } from 'common/types'
import { fontSize } from 'frontend/styles/common'
import MessageResult from './MessageResult'

type Props = {
  message: Message,
  isPinned: boolean
}

const MessageContainer = styled.div`
  padding: 0.5rem 0;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  display: inline-block;
  width: 100%;

  &:first-child {
    margin-top: auto;
  }
`

const From = styled.div`
  font-weight: bold;
  font-size: ${fontSize.small};
  color: ${props => props.theme.colorSecondary};
  padding: 0.2rem 1rem 0;
`

const Sent = styled.span`
  color: #999;
  font-weight: normal;
  font-style: italic;
`

const Text = styled.div`
  padding: 0 1rem 0.2rem;
  color: ${props => props.theme.color};
`

export default class MessageView extends React.Component<Props> {
  render() {
    const { isPinned } = this.props
    const { from, timestamp, text, result } = this.props.message
    const hoverText = new Date(timestamp)
    const formattedDate = new Timeago().format(timestamp)
    return (
      <MessageContainer isPinned={isPinned}>
        <From>
          {from} <Sent title={hoverText}>{formattedDate}</Sent>
        </From>
        <Text>{text}</Text>
        <MessageResult result={result} />
      </MessageContainer>
    )
  }
}
