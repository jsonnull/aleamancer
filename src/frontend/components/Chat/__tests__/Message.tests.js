// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Message from '../Message.js'

describe('Chat Message component', () => {
  it('renders correctly', () => {
    const message = {
      id: 'key',
      from: 'testUser',
      text: 'text',
      result: null,
      timestamp: new Date(0)
    }
    const tree = renderer
      .create(<Message message={message} isPinned={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
