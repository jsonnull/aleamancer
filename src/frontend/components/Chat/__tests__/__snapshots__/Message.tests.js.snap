// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Message component renders correctly 1`] = `
.c0 {
  padding: 0.5rem 0;
  border-bottom: 1px solid;
  display: inline-block;
  width: 100%;
}

.c0:first-child {
  margin-top: auto;
}

.c1 {
  font-weight: bold;
  font-size: 1.3rem;
  padding: 0.2rem 1rem 0;
}

.c2 {
  color: #999;
  font-weight: normal;
  font-style: italic;
}

.c3 {
  padding: 0 1rem 0.2rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    testUser
     
    <span
      className="c2"
      title={1970-01-01T00:00:00.000Z}
    >
      48 years ago
    </span>
  </div>
  <div
    className="c3"
  >
    text
  </div>
</div>
`;
